-1

Tested at: 770e744b5b82f67a70148a6ffd645161cd21298f

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-26056/33383/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  ClangBuild**

* **Build**:

I found compilation error when building: 

<pre>>> Creating project symlinks
>> Entering Package SimG4Core/PhysicsLists
>> Entering Package src/BigProducts/Simulation
>> Entering Package src/BigProducts/Simulation
>> Building big plugin tmp/slc6_amd64_gcc700/src/BigProducts/Simulation/pluginSimulation.so
c++: error: objs/slc6_amd64_gcc700/SimG4CorePhysicsLists.obj: No such file or directory
gmake: *** [tmp/slc6_amd64_gcc700/src/BigProducts/Simulation/pluginSimulation.so] Error 1
>> Leaving Package src/BigProducts/Simulation
>> Package src/BigProducts/Simulation built
>> Leaving Package src/BigProducts/Simulation
>> Package src/BigProducts/Simulation built
</pre>

* **Clang**:

I found compilation error while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 32 COMPILER='llvm compile'

```


<pre>>> Compiling  /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2019-03-04-1100/src/SimG4Core/PhysicsLists/src/MonopoleTransportation.cc 
>> Compiling  /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2019-03-04-1100/src/SimG4Core/PhysicsLists/src/CMSFTFPNeutronBuilder.cc 
>> Compiling  /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2019-03-04-1100/src/SimG4Core/PhysicsLists/src/CMSFTFPProtonBuilder.cc 
>> Compiling  /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2019-03-04-1100/src/SimG4Core/PhysicsLists/src/DummyEMPhysics.cc 
>> Compiling  /build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2019-03-04-1100/src/SimG4Core/PhysicsLists/src/UrbanMscModel93.cc 
/build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2019-03-04-1100/src/SimG4Core/PhysicsLists/src/CMSMonopolePhysics.cc:16:21: error: out-of-line definition of 'CMSMonopolePhysics' does not match any declaration in 'CMSMonopolePhysics'
CMSMonopolePhysics::CMSMonopolePhysics(const HepPDT::ParticleDataTable * pdt,
                    ^~~~~~~~~~~~~~~~~~
/build/cmsbld/jenkins/workspace/ib-any-integration/CMSSW_10_2_X_2019-03-04-1100/src/SimG4Core/PhysicsLists/src/CMSMonopolePhysics.cc:114:34: error: no matching constructor for initialization of 'MonopoleTransportation'
        pmanager->AddProcess(new MonopoleTransportation(mpl,verbose),-1,0,0);
                                 ^                      ~~~~~~~~~~~
</pre>

--- a/SimG4Core/PhysicsLists/src/CMSMonopolePhysics.cc
+++ b/SimG4Core/PhysicsLists/src/CMSMonopolePhysics.cc
@@ -1,6 +1,7 @@
 #include "SimG4Core/PhysicsLists/interface/CMSMonopolePhysics.h"
 #include "SimG4Core/PhysicsLists/interface/MonopoleTransportation.h"
 #include "SimG4Core/PhysicsLists/interface/CMSmplIonisation.h"
+#include "SimG4Core/MagneticField/interface/ChordFinderSetter.h"
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
 
 #include "G4ParticleDefinition.hh"
@@ -14,9 +15,10 @@
 #include "CLHEP/Units/GlobalSystemOfUnits.h"
 
 CMSMonopolePhysics::CMSMonopolePhysics(const HepPDT::ParticleDataTable * pdt,
+				       sim::ChordFinderSetter * cfs, 
 				       const edm::ParameterSet & p) :
-  G4VPhysicsConstructor("Monopole Physics")
-{  
+  G4VPhysicsConstructor("Monopole Physics"), chordFinderSetter(cfs) {
+  
   verbose   = p.getUntrackedParameter<int>("Verbosity",0);
   magCharge = p.getUntrackedParameter<int>("MonopoleCharge",1);
   deltaRay  = p.getUntrackedParameter<bool>("MonopoleDeltaRay",true);
@@ -94,8 +96,8 @@ void CMSMonopolePhysics::ConstructProcess() {
       if(!pmanager) {
         std::ostringstream o;
         o << "Monopole without a Process Manager";
-        throw edm::Exception( edm::errors::Configuration, o.str().c_str());
-        return;
+        G4Exception("CMSMonopolePhysics::ConstructProcess()","",
+                    FatalException,o.str().c_str());
       }
 
       G4double magn = mpl->MagneticCharge();
@@ -109,9 +111,9 @@ void CMSMonopolePhysics::ConstructProcess() {
       }
   
       if (magn != 0.0) {
-        G4int idxt(0);
-        pmanager->RemoveProcess(idxt);
-        pmanager->AddProcess(new MonopoleTransportation(mpl,verbose),-1,0,0);
+	pmanager->RemoveProcess(0);
+	pmanager->AddProcess(new MonopoleTransportation(mpl,chordFinderSetter,verbose),
+                                                        -1, 0, 0);
       }
 
       if (mpl->GetPDGCharge() != 0.0) {

--- a/SimG4Core/PhysicsLists/src/CMSMonopolePhysics.cc
+++ b/SimG4Core/PhysicsLists/src/CMSMonopolePhysics.cc
@@ -11,6 +11,7 @@
 #include "G4hMultipleScattering.hh"
 #include "G4hIonisation.hh"
 #include "G4hhIonisation.hh"
+#include "G4mplIonisationModel.hh"
 
 #include "CLHEP/Units/GlobalSystemOfUnits.h"
 
@@ -126,6 +127,11 @@ void CMSMonopolePhysics::ConstructProcess() {
       }
       if(magn != 0.0) {
 	CMSmplIonisation* mplioni = new CMSmplIonisation(magn);
+        if(!deltaRay) {
+          G4mplIonisationModel* ion = new G4mplIonisationModel(magn,"PAI");
+          ion->SetParticle(mpl);
+          mplioni->AddEmModel(0,ion,ion);
+	}
 	ph->RegisterProcess(mplioni, mpl);
       }
       pmanager->AddDiscreteProcess(new G4StepLimiter());

